server/src/main/java/com/octopus/teamcity/opentelemetry/server/BuildOverviewExtensionController.java

```diff
@@ -0,0 +1,126 @@
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.util.Date;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+import static com.octopus.teamcity.opentelemetry.common.PluginConstants.*;
+
+public class BuildOverviewExtensionController extends BaseController
+{
+    private static final Logger LOG = Logger.getLogger(BuildOverviewExtensionController.class.getName());
+
+    private final SBuildServer sBuildServer;
+    private final ProjectManager projectManager;
+    private final PluginDescriptor pluginDescriptor;
+    private BuildStorageManager buildStorageManager;
+
+    public BuildOverviewExtensionController(
+            @NotNull PagePlaces pagePlaces,
+            @NotNull SBuildServer sBuildServer,
+            @NotNull ProjectManager projectManager,
+            @NotNull PluginDescriptor pluginDescriptor,
+            @NotNull WebControllerManager controllerManager,
+            @NotNull BuildStorageManager buildStorageManager)
+    {
+        this.sBuildServer = sBuildServer;
+        this.projectManager = projectManager;
+
+        this.pluginDescriptor = pluginDescriptor;
+        this.buildStorageManager = buildStorageManager;
+
+        String url = "/otel-trace-url.html";
+
+        final SimplePageExtension sakuraPageExtension = new SimplePageExtension(pagePlaces);
+        sakuraPageExtension.setPluginName(PLUGIN_NAME);
+        sakuraPageExtension.setPlaceId(new PlaceId("SAKURA_BUILD_OVERVIEW"));
+        sakuraPageExtension.setIncludeUrl(url);
+        sakuraPageExtension.register();
+
+        // we're not showing on the classic UI, as we want it to appear on the build summary
+        // but if we do that, it appears twice on the Sakura UI, and it's in an extremely ugly
+        // iframe that makes me cry
+
+        //todo: reach out to JetBrains and ask about it
+
+        // final SimplePageExtension classicPageExtension = new SimplePageExtension(pagePlaces);
+        // classicPageExtension.setPluginName(PLUGIN_NAME);
+        // classicPageExtension.setPlaceId(PlaceId.BUILD_RESULTS_FRAGMENT);
+        // classicPageExtension.setIncludeUrl(url);
+        // classicPageExtension.register();
+
+        controllerManager.registerController(url, this);
+    }
+
+    @Nullable
+    @Override
+    protected ModelAndView doHandle(@NotNull HttpServletRequest request, @NotNull HttpServletResponse response) throws Exception {
+        final ModelAndView mv = new ModelAndView(pluginDescriptor.getPluginResourcesPath("buildOverviewExtension.jsp"));
+        boolean isSakuraUI = WebUtil.sakuraUIOpened(request);
+
+        //todo: I think we need to do some auto-refreshes until we have data?
+
+        Long buildId;
+
+        if (isSakuraUI) {
+            PluginUIContext pluginUIContext = PluginUIContext.getFromRequest(request);
+            buildId = pluginUIContext.getBuildId();
+        } else {
+            buildId = Long.parseLong(request.getParameter("buildId"));
+        }
```

Comment:
I believe this can be simplified to something like 
```
Long buildId = WebUtil.sakuraUIOpened(request) ? PluginUIContext.getFromRequest(request).getBuildId() : Long.parseLong(request.getParameter("buildId"));
```

But also for easier reading the variables make sense. Options!